process.c

```diff
@@ -8639,38 +8639,108 @@ get_PROCESS_ID(ID _x, VALUE *_y)
 
 /*
  *  call-seq:
- *     Process.kill(signal, pid, *pids)    -> integer
- *
- *  Sends the given signal to the specified process id(s) if _pid_ is positive.
- *  If _pid_ is zero, _signal_ is sent to all processes whose group ID is equal
- *  to the group ID of the process. If _pid_ is negative, results are dependent
- *  on the operating system. _signal_ may be an integer signal number or
- *  a POSIX signal name (either with or without a +SIG+ prefix). If _signal_ is
- *  negative (or starts with a minus sign), kills process groups instead of
- *  processes. Not all signals are available on all platforms.
- *  The keys and values of Signal.list are known signal names and numbers,
- *  respectively.
- *
- *     pid = fork do
- *        Signal.trap("HUP") { puts "Ouch!"; exit }
- *        # ... do some work ...
- *     end
- *     # ...
- *     Process.kill("HUP", pid)
- *     Process.wait
+ *    Process.kill(signal, *ids) -> count
+ *
+ *  Sends a signal to each process specified by +ids+
+ *  (which must specify at least one ID);
+ *  returns the count of signals sent.
+ *
+ *  For each given +id+, if +id+ is:
+ *
+ *  - Positive, sends the signal to the process whose process ID is +id+.
+ *  - Zero, send the signal to all processes in the current process group.
+ *  - Negative, sends the signal to a system-dependent collection of processes.
+ *
+ *  Argument +signal+ specifies the signal to be sent;
+ *  the argument may be:
+ *
+ *  - An integer signal number: e.g., +-29+, +0+, +29+.
+ *  - A signal name (string), with or without leading <tt>'SIG'</tt>,
+ *    and with or without a further prefixed minus sign (<tt>'-'</tt>):
+ *    e.g.:
+ *
+ *    - <tt>'SIGPOLL'</tt>.
+ *    - <tt>'POLL'</tt>,
+ *    - <tt>'-SIGPOLL'</tt>.
+ *    - <tt>'-POLL'</tt>.
+ *
+ *  - A signal symbol, with or without leading <tt>'SIG'</tt>,
+ *    and with or without a further prefixed minus sign (<tt>'-'</tt>):
+ *    e.g.:
+ *
+ *    - +:SIGPOLL+.
+ *    - +:POLL+.
+ *    - <tt>:'-SIGPOLL'</tt>.
+ *    - <tt>:'-POLL'</tt>.
+ *
+ *  If +signal+ is:
+ *
+ *  - A non-negative integer, or a signal name or symbol
+ *    without prefixed <tt>'-'</tt>,
+ *    each process with process ID +id+ is signalled.
+ *  - A negative integer, or a signal name or symbol
+ *    with prefixed <tt>'-'</tt>,
+ *    each process group with group ID +id+ is signalled.
+ *
+ *  The signal non-negative integers and strings are:
+ *
+ *  - +0+ (<tt>'SIGEXIT'</tt>): Exit the current process.
```

Comment:
This is not a real signal, nor exit the process.
- For `Process.kill`, `0` examines if the destination process is alive or not, but `SIGEXIT` is nog recognized as a signal.
- For `Signal.trap`, Hook for when the current process exits.

Reply:
Ditto,